### ISSUE TITLE
Elect Lodash Methods To Demystify ~ Log

### ISSUE THREAD

**Issue by kingdavidmartins** [Chalarangelo/30-seconds-of-code#100]
Hey @elderhsouza Thanks again for the awesome log suggestion 👋  & Hey All 😄 

## Library

**Category**

- [x] link to snippet

**Legend**
- [x] link to snippet ~ Means no PR has been ref or Submitted
- [x] link to snippet | PR ref ~ Means PR is opened and ref
- [x] <del> link to snippet | PR ref </del> ~ Means PR was closed and not merged
- [x] link to snippet | `file-name.md` ~ Means PR has been merged & snippet has been implemented 

**Note** *If a snippet does not have a PR attached. That means a PR wasn't created or referenced*

Here's a list of the following snippets so far with their respective PR. Remember if there isn't a PR ref attached to a snippet it means no one has gotten around to it yet 💔 Which also means there's work to be done & contributions to be made 💻 💯  

Don't be shy and give it a try. Fork, Clone, Contribute, & Submit a PR 🎊 

Also shout out to @Chalarangelo for starting such an awesome project 🎉 

### *REMEMBER*
Follow the guidelines in CONTRIBUTING.md
No copy-pasting. Let's just try to have a fun and educative experience

## Lodash

**Function**

- [x]  https://lodash.com/docs/4.17.4#throttle | `throttle.md`
- [x]  https://lodash.com/docs/4.17.4#debounce | `debounce.md`
- [x]  https://lodash.com/docs/4.17.4#rearg | `rearg.md`
- [x]  https://lodash.com/docs/4.17.4#overArgs | `overArgs.md`
- [x]  https://lodash.com/docs/4.17.4#rest | `collectInto.md`
- [x]  https://lodash.com/docs/4.17.4#partial | `partial.md`
- [x]  https://lodash.com/docs/4.17.4#partialRight | `partialRight.md`
- [x]  https://lodash.com/docs/4.17.4#delay | `delay.md`
- [x]  https://lodash.com/docs/4.17.4#bindKey | `bindKey.md`
- [x]  https://lodash.com/docs/4.17.4#bind | `bind.md`
- [x]  https://lodash.com/docs/4.17.4#ary | `ary.md`
- [x]  https://lodash.com/docs/4.17.4#unary | `unary.md`
- [x]  https://lodash.com/docs/4.17.4#negate | `negate.md`
- [x]  https://lodash.com/docs/4.17.4#once | #455
- [x]  https://lodash.com/docs/4.17.4#defer | #454
- [x]  https://lodash.com/docs/4.17.4#memoize | #433
- [x]  https://lodash.com/docs/4.17.4#wrap
- [x]  https://lodash.com/docs/4.17.4#after | `times.md`
- [x]  https://lodash.com/docs/4.17.4#before | `times.md`
- [x]  https://lodash.com/docs/4.17.4#spread | `spreadOver.md`
- [x]  https://lodash.com/docs/4.17.4#flip | `flip.md`
- [x]  https://lodash.com/docs/4.17.4#curry | `curry.md`
- [x]  https://lodash.com/docs/4.17.4#curryRight | `pipe.md`

**Util**

- [x] https://lodash.com/docs/4.17.4#attempt | `attempt.md`
- [x] ~~https://lodash.com/docs/4.17.4#conforms~~
- [x] https://lodash.com/docs/4.17.4#bindAll | `bindAll.md`
- [x] https://lodash.com/docs/4.17.4#flow | `compose.md`
- [x] https://lodash.com/docs/4.17.4#flowRight | `composeRight.md`
- [x] ~~https://lodash.com/docs/4.17.4#iteratee~~
- [x] ~~https://lodash.com/docs/4.17.4#method~~
- [x] ~~https://lodash.com/docs/4.17.4#methodOf~~
- [x] ~~https://lodash.com/docs/4.17.4#mixin~~
- [x] ~~https://lodash.com/docs/4.17.4#matchesProperty~~
- [x] ~~https://lodash.com/docs/4.17.4#noConflict~~
- [x] https://lodash.com/docs/4.17.4#nthArg | `nthArg.md`
- [x] https://lodash.com/docs/4.17.4#over | `over.md`
- [x] ~~https://lodash.com/docs/4.17.4#toPath~~
- [x] https://lodash.com/docs/4.17.4#matches | `matches.md`
- [x] ~~https://lodash.com/docs/4.17.4#times~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#identity~~
- [x] https://lodash.com/docs/4.17.4#uniqueId | `UUIDGeneratorBrowser.md` & `UUIDGeneratorNode.md`
- [x] ~~https://lodash.com/docs/4.17.4#stubArray~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#stubFalse~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#stubObject~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#stubString~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#stubTrue~~ | Natively implemented
- [x] https://lodash.com/docs/4.17.4#range | `initializeArrayWithRange.md`
- [x] https://lodash.com/docs/4.17.4#rangeRight | `initializeArrayWithRangeRight.md`
- [x] https://lodash.com/docs/4.17.4#property
- [x] https://lodash.com/docs/4.17.4#propertyOf | `select.md`
- [x] ~~https://lodash.com/docs/4.17.4#overEvery~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#overSome~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#noop~~
- [x] ~~https://lodash.com/docs/4.17.4#runInContext~~
- [x] ~~https://lodash.com/docs/4.17.4#defaultTo~~
- [x] ~~https://lodash.com/docs/4.17.4#constant~~
- [x] ~~https://lodash.com/docs/4.17.4#cond~~

**Array**

- [x]  https://lodash.com/docs/4.17.4#pullAllBy | `pullBy.md`
- [x]  ~~https://lodash.com/docs/4.17.4#pullAllWith~~
- [x]  https://lodash.com/docs/4.17.4#sortedIndexBy | `sortedIndexBy.md`
- [x]  https://lodash.com/docs/4.17.4#sortedLastIndexBy | `sortedLastIndexBy.md`
- [x]  https://lodash.com/docs/4.17.4#takeRightWhile | `takeRightWhile.md`
- [x]  https://lodash.com/docs/4.17.4#takeWhile | `takeWhile.md`
- [x]  https://lodash.com/docs/4.17.4#dropRightWhile | `dropRightWhile.md`
- [x]  https://lodash.com/docs/4.17.4#dropWhile | `dropWhile.md`
- [x]  ~~https://lodash.com/docs/4.17.4#sortedIndexOf~~
- [x]  ~~https://lodash.com/docs/4.17.4#sortedLastIndexOf~~
- [x]  https://lodash.com/docs/4.17.4#sortedLastIndex | `sortedLastIndex.md`
- [x]  ~~https://lodash.com/docs/4.17.4#sortedUniq~~
- [x]  ~~https://lodash.com/docs/4.17.4#sortedUniqBy~~
- [x]  https://lodash.com/docs/4.17.4#intersectionBy | `intersectionBy.md`
- [x]  https://lodash.com/docs/4.17.4#intersectionWith | `intersectionWith.md`
- [x]  https://lodash.com/docs/4.17.4#unzipWith | `unzipWith.md`
- [x]  https://lodash.com/docs/4.17.4#unzip | `unzip.md`
- [x]  https://lodash.com/docs/4.17.4#unionBy | `unionBy.md`
- [x]  https://lodash.com/docs/4.17.4#unionWith | `unionWith.md`
- [x]  https://lodash.com/docs/4.17.4#xorBy | `symmetricDifferenceBy.md`
- [x]  https://lodash.com/docs/4.17.4#xorWith | `symmetricDifferenceWith.md`
- [x]  https://lodash.com/docs/4.17.4#differenceBy | `differenceBy.md`
- [x]  https://lodash.com/docs/4.17.4#differenceWith | `differenceWith .md`
- [x]  https://lodash.com/docs/4.17.4#zipWith | `zipWith.md`
- [x]  https://lodash.com/docs/4.17.4#sortedIndex | #440
- [x]  https://lodash.com/docs/4.17.4#xor | `symmetricDifference.md`
- [x]  https://lodash.com/docs/4.17.4#dropRight
- [x]  https://lodash.com/docs/4.17.4#zipObject
- [x]  https://lodash.com/docs/4.17.4#pullAll | `pull.md` 
- [x]  https://lodash.com/docs/4.17.4#pullAt | `pullAtIndex.md`
- [x]  https://lodash.com/docs/4.17.4#pull | `pull.md`
- [x]  https://lodash.com/docs/4.17.4#remove | `remove.md`
- [x]  https://lodash.com/docs/4.17.4#without | `without.md`
- [x]  https://lodash.com/docs/4.17.4#takeRight | `takeRight.md`
- [x]  https://lodash.com/docs/4.17.4#zip | `zip.md`
- [x]  https://lodash.com/docs/4.17.4#intersection | `intersection.md`
- [x]  https://lodash.com/docs/4.17.4#flattenDepth | `flatten-array-up-to-depth.md`
- [x]  https://lodash.com/docs/4.17.4#nth | `nth-element-of-array.md` 
- [x]  https://lodash.com/docs/4.17.4#fill | #118
- [x]  https://lodash.com/docs/4.17.4#drop | `drop.md`
- [x]  https://lodash.com/docs/4.17.4#flatten | `flatten-array.md`
- [x]  https://lodash.com/docs/4.17.4#flattenDeep | `deep-flatten-array.md`
- [x]  https://lodash.com/docs/4.17.4#union | `union.md`
- [x]  https://lodash.com/docs/4.17.4#chunk | `chunk.md` 
- [x]  https://lodash.com/docs/4.17.4#compact | `compact.md`
- [x]  https://lodash.com/docs/4.17.4#concat | #107
- [x]  https://lodash.com/docs/4.17.4#difference | `difference.md` 
- [x]  https://lodash.com/docs/4.17.4#head | `head.md`
- [x]  https://lodash.com/docs/4.17.4#fromPairs | `objectFromPairs.md`
- [x]  https://lodash.com/docs/4.17.4#last | `last.md` 
- [x]  https://lodash.com/docs/4.17.4#initial | `initial.md`
- [x]  https://lodash.com/docs/4.17.4#tail | `tail.md` 
- [x]  https://lodash.com/docs/4.17.4#take | `take.md` 
- [x]  ~~https://lodash.com/docs/4.17.4#zipObjectDeep~~ | `zipObject` is sufficient for most use-cases
- [x] ~~https://lodash.com/docs/4.17.4#findIndex~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#findLastIndex~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#join~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#indexOf~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#lastIndexOf~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#reverse~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#slice~~  | Natively implemented

**Lang**

- [x]  https://lodash.com/docs/4.17.4#castArray | `castArray.md`
- [x]  https://lodash.com/docs/4.17.4#cloneDeep | `deepClone.md`
- [x]  ~~https://lodash.com/docs/4.17.4#cloneDeepWith~~
- [x]  ~~https://lodash.com/docs/4.17.4#cloneWith~~
- [x]  ~~https://lodash.com/docs/4.17.4#conformsTo~~
- [x]  https://lodash.com/docs/4.17.4#isMatch | `matches.md`
- [x]  https://lodash.com/docs/4.17.4#isMatchWith | `matchesWith.md`
- [x]  ~~https://lodash.com/docs/4.17.4#isSafeInteger~~ | Natively implemented
- [x]  https://lodash.com/docs/4.17.4#isObjectLike | `isObjectLike.md`
- [x]  https://lodash.com/docs/4.17.4#isEmpty | `isEmpty.md`
- [x]  https://lodash.com/docs/4.17.4#isPlainObject | `isPlainObject.md`
- [x]  ~~https://lodash.com/docs/4.17.4#isNative~~ | Unstable, unreliable, check lodash's note
- [x]  https://lodash.com/docs/4.17.4#isUndefined | `isUndefined .md`
- [x]  https://lodash.com/docs/4.17.4#isError | `is.md`
- [x]  https://lodash.com/docs/4.17.4#isBuffer | `is.md`
- [x]  https://lodash.com/docs/4.17.4#isElement | `is.md`
- [x]  https://lodash.com/docs/4.17.4#isInteger | `is.md`
- [x]  https://lodash.com/docs/4.17.4#isNil | `isNil.md`
- [x]  ~~https://lodash.com/docs/4.17.4#isArguments~~
- [x]  ~~https://lodash.com/docs/4.17.4#isLength~~
- [x]  https://lodash.com/docs/4.17.4#isArrayBuffer | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isMap | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isRegExp | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isSet | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isTypedArray | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isWeakMap | `is .md`
- [x]  https://lodash.com/docs/4.17.4#isWeakSet | `is.md`
- [x]  ~~https://lodash.com/docs/4.17.4#isArrayLikeObject~~
- [x]  https://lodash.com/docs/4.17.4#isEqual | #546
- [x]  ~~https://lodash.com/docs/4.17.4#isEqualWith~~
- [x]  ~~https://lodash.com/docs/4.17.4#toFinite~~
- [x]  ~~https://lodash.com/docs/4.17.4#toLength~~
- [x] ~~https://lodash.com/docs/4.17.4#toNumber~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#toPlainObject~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#toArray~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#toInteger~~ | Natively implemented
- [x]  https://lodash.com/docs/4.17.4#toSafeInteger | `toSafeInteger.md`
- [x]  https://lodash.com/docs/4.17.4#isObject | `isPrimitive.md`
- [x]  https://lodash.com/docs/4.17.4#isArrayLike | #438 
- [x]  ~~https://lodash.com/docs/4.17.4#isFinite~~ | Natively implemented
- [x] https://lodash.com/docs/4.17.4#isNull
- [x]  ~~https://lodash.com/docs/4.17.4#toString~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#isNaN~~ | Natively implemented
- [x]  https://lodash.com/docs/4.17.4#clone | `shallowClone.md`
- [x]  https://lodash.com/docs/4.17.4#isArray | `isArray.md`
- [x]  https://lodash.com/docs/4.17.4#isBoolean | `isBoolean.md`
- [x]  https://lodash.com/docs/4.17.4#isFunction | `isFunction.md`
- [x]  https://lodash.com/docs/4.17.4#isNumber | `isNumber.md`
- [x]  https://lodash.com/docs/4.17.4#isString | `isString.md`
- [x]  https://lodash.com/docs/4.17.4#isSymbol | `isSymbol.md`
- [x]  ~~https://lodash.com/docs/4.17.4#lt~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#lte~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#eq~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#gt~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#gte~~ | Natively implemented

**Object**

- [x] ~~https://lodash.com/docs/4.17.4#defaultsDeep~~
- [x] https://lodash.com/docs/4.17.4#findKey | `findKey.md`
- [x] https://lodash.com/docs/4.17.4#findLastKey | `findLastKey.md`
- [x] ~~https://lodash.com/docs/4.17.4#forIn~~
- [x] ~~https://lodash.com/docs/4.17.4#forInRight~~
- [x] https://lodash.com/docs/4.17.4#defaults | `defaults.md`
- [x] ~~https://lodash.com/docs/4.17.4#invoke~~
- [x] ~~https://lodash.com/docs/4.17.4#result~~
- [x] https://lodash.com/docs/4.17.4#omitBy | `omitBy.md`
- [x] https://lodash.com/docs/4.17.4#pickBy | `pickBy.md`
- [x] https://lodash.com/docs/4.17.4#omit | `omit.md`
- [x] ~~https://lodash.com/docs/4.17.4#set~~
- [x] ~~https://lodash.com/docs/4.17.4#setWith~~
- [x] ~~https://lodash.com/docs/4.17.4#unset~~
- [x] ~~https://lodash.com/docs/4.17.4#update~~
- [x] ~~https://lodash.com/docs/4.17.4#updateWith~~
- [x] https://lodash.com/docs/4.17.4#get | `get.md`
- [x] https://lodash.com/docs/4.17.4#forOwnRight | #559
- [x] https://lodash.com/docs/4.17.4#forOwn | #559
- [x] ~~https://lodash.com/docs/4.17.4#mergeWith~~
- [x] ~~https://lodash.com/docs/4.17.4#assignInWith~~
- [x] https://lodash.com/docs/4.17.4#merge | `merge.md`
- [x] https://lodash.com/docs/4.17.4#transform | `transform.md`
- [x] https://lodash.com/docs/4.17.4#functions | `functions.md`
- [x] https://lodash.com/docs/4.17.4#functionsIn | `functions.md`
- [x] https://lodash.com/docs/4.17.4#mapKeys | `mapKeys.md`
- [x] https://lodash.com/docs/4.17.4#mapValues | `mapValues.md`
- [x] https://lodash.com/docs/4.17.4#at | `select.md`
- [x] https://lodash.com/docs/4.17.4#pick | `pick.md`
- [x] https://lodash.com/docs/4.17.4#invert | `invertKeyValues.md`
- [x] https://lodash.com/docs/4.17.4#invertBy | `invertKeyValues.md`
- [x] ~~https://lodash.com/docs/4.17.4#hasIn~~
- [x] ~~https://lodash.com/docs/4.17.4#has~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#valuesIn~~
- [x] ~~https://lodash.com/docs/4.17.4#values~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#keysIn~~
- [x] ~~https://lodash.com/docs/4.17.4#keys~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#toPairsIn~~
- [x] https://lodash.com/docs/4.17.4#toPairs | `objectToPairs.md`
- [x] ~~https://lodash.com/docs/4.17.4#create~~ | Natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#assignIn~~
- [x] ~~https://lodash.com/docs/4.17.4#assign~~ | Natively implemented

**Collection**

- [x]  https://lodash.com/docs/4.17.4#findLast | `findLast.md`
- [x]  https://lodash.com/docs/4.17.4#countBy | `countBy.md`
- [x]  ~~https://lodash.com/docs/4.17.4#invokeMap~~ | Natively implemented
- [x]  https://lodash.com/docs/4.17.4#keyBy | `groupBy.md`
- [x]  ~~https://lodash.com/docs/4.17.4#flatMap~~
- [x]  ~~https://lodash.com/docs/4.17.4#flatMapDeep~~
- [x]  ~~https://lodash.com/docs/4.17.4#flatMapDepth~~
- [x]  https://lodash.com/docs/4.17.4#forEachRight | `forEachRight.md`
- [x]  https://lodash.com/docs/4.17.4#groupBy | `groupBy.md`
- [x]  https://lodash.com/docs/4.17.4#partition | #512 
- [x]  ~~https://lodash.com/docs/4.17.4#reject~~ | #511 
- [x]  https://lodash.com/docs/4.17.4#sortBy | `orderBy.md`
- [x]  https://lodash.com/docs/4.17.4#sampleSize | #434
- [x]  https://lodash.com/docs/4.17.4#size | #427
- [x]  https://lodash.com/docs/4.17.4#sample | `sample.md`
- [x]  https://lodash.com/docs/4.17.4#shuffle | `shuffle.md`
- [x]  https://lodash.com/docs/4.17.4#orderBy | `orderBy.md`
- [x]  ~~https://lodash.com/docs/4.17.4#includes~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#every~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#find~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#some~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#forEach~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#filter~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#reduce~~ | Natively implemented
- [x]  https://lodash.com/docs/4.17.4#reduceRight
- [x]  ~~https://lodash.com/docs/4.17.4#map~~ | Natively implemented

**Math**

- [x]  https://lodash.com/docs/4.17.4#maxBy | `maxBy.md`
- [x]  https://lodash.com/docs/4.17.4#minBy | `minBy.md`
- [x]  https://lodash.com/docs/4.17.4#meanBy | `averageBy.md`
- [x]  https://lodash.com/docs/4.17.4#sumBy | `sumBy.md`
- [x]  https://lodash.com/docs/4.17.4#sum | `arraySum.md`
- [x] https://lodash.com/docs/4.17.4#max | `arrayMax.md`
- [x]  https://lodash.com/docs/4.17.4#mean | `arrayAverage.md`
- [x]  ~~https://lodash.com/docs/4.17.4#ceil~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#multiply~~ | natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#add~~ | natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#divide~~ | natively implemented
- [x] ~~https://lodash.com/docs/4.17.4#floor~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#round~~  | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#subtract~~ | natively implemented
  
**String**

- [x]  ~~https://lodash.com/docs/4.17.4#deburr~~ | Requires lookup table, will not implement
- [X]  https://lodash.com/docs/4.17.4#lowerFirst | `decapitalize.md`
- [x]  ~~https://lodash.com/docs/4.17.4#startCase~~
- [x]  ~~https://lodash.com/docs/4.17.4#repeat~~ | Natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#template~~ | Natively implemented (template literals)
- [x]  https://lodash.com/docs/4.17.4#escape | `escapeHTML.md`
- [x]  https://lodash.com/docs/4.17.4#unescape | `unescapeHTML.md`
- [x] ~~https://lodash.com/docs/4.17.4#pad~~ | Natively implemented 
- [x]  https://lodash.com/docs/4.17.4#kebabCase | `toKebabCase.md`
- [x]  https://lodash.com/docs/4.17.4#snakeCase | #321
- [x]  https://lodash.com/docs/4.17.4#words | `words.md`
- [x]  https://lodash.com/docs/4.17.4#escapeRegExp | `escapeRegExpmd`
- [x]  https://lodash.com/docs/4.17.4#camelCase | `toCamelCase.md`
- [x]  https://lodash.com/docs/4.17.4#capitalize | `capitalizeEveryWord.md`
- [x]  https://lodash.com/docs/4.17.4#upperFirst | `capitalize.md`
- [x]  https://lodash.com/docs/4.17.4#truncate | `truncate.md`
- [x]  ~~https://lodash.com/docs/4.17.4#lowerCase~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#padEnd~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#padStart~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#parseInt~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#replace~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#endsWith~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#split~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#startsWith~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#toLower~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#toUpper~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#trim~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#trimEnd~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#trimStart~~ | natively implemented
- [x]  ~~https://lodash.com/docs/4.17.4#upperCase~~ | natively implemented 
  
**Number**

- [x] https://lodash.com/docs/4.17.4#clamp | `clampNumber.md`
- [x] https://lodash.com/docs/4.17.4#inRange | `inRange.md`
- [x] https://lodash.com/docs/4.17.4#random | `randomNumberInRange.md`

**Date**

- [x]  https://lodash.com/docs/4.17.4#now

**Comment by Chalarangelo** [comment]
@kingdavidmartins I love you, man. I will start working through a few of these over time. This list is awesome. Hope we see it to completion eventually!

**Comment by meetzaveri** [comment]
So should we start picking one by one from these resource? Is that you saying @Chalarangelo

**Comment by Chalarangelo** [comment]
@meetzaveri yep, let's ge started. Follow the guidelines in CONTRIBUTING.md to make merging PRs easier, work on individual files and state here which ones you intend to write (even though you might not), so we don't end up with duplicates if possible. I am clearing up PRs right now, will try to chip through some of the methods listed here later today.

**Comment by sabareeshkk** [comment]
@Chalarangelo  i am excited to try any one of these.

**Comment by kingdavidmartins** [comment]
@sabareeshkk Yea give some of them a shot. Hope you submit a PR 😄

**Comment by sabareeshkk** [comment]
@kingdavidmartins sure will do :+1:

**Comment by Chalarangelo** [comment]
@kingdavidmartins I think we have [https://lodash.com/docs/4.17.4#difference](https://lodash.com/docs/4.17.4#difference) implemented under `array-difference.md`.

**Comment by kingdavidmartins** [comment]
@Chalarangelo if there is any snippets I am missing please let me know. I kinda was up all night so not as sharp as I should be. Later on after some rest I should have all the current snippets that have been implemented updated  and extend the list to include the Collection, Lang, Date, and Function section/category

**Comment by Chalarangelo** [comment]
Alright, I'll see if we have any of them and I will also get coding now that the PR list is almost empty.

**Comment by Chalarangelo** [comment]
I updated some the list with some of the changes made and implementations added, also moved all completed functions to the very bottom to be easier to find new things to implement at first glance.

**Comment by Chalarangelo** [comment]
By the way I am sticking to the same PR procedure as everybody else for at least some of these snippets, so that I can get feedback before merging them in. Otherwise, we can easily lose track of what's going on.

**Comment by Chalarangelo** [comment]
I believe lodash's `slice` and `reverse` are pretty much polyfills for older browsers. We should not implement those.

**Comment by Chalarangelo** [comment]
@kingdavidmartins I tried to clean up the log by marking the natively implemented functions, as well as the ones I remember that we have implemented already. We seem to be about 30% through it, which is very impressive for just one week!

**Comment by fejes713** [comment]
@kingdavidmartins & @Chalarangelo are we going to create a library out of this? 😄

**Comment by Chalarangelo** [comment]
@feiwen8772 In all honesty, I don't know. When we're done with Lodash's implementations and the PRs stop coming in so fast, we should look into packaging everything, seeing if people want that. For now, it's a bit out of the scope of the project and there's so much else that needs to be done that we can put that idea on hold for the next month or so. But it keeps coming up all the time, so we will eventually have to discuss that.

**Comment by skatcat31** [comment]
So something that's been bothering me: why are we demystifying lodash?  
We aren't, nor should we try to be, a replacement. We are more a place for people to learn cool tricks in small snippets aren't we?  
IFF we wanted to be a library we would need to be MUCH more structured about our approach to development, and pull requests, and implimentation( ala Ramda, Lodash, and other libraries that allow community support )  
If we want to do such a series, why would we do it here? It'll add a lot of previous arts to questionable gain IMHO.

**Comment by kingdavidmartins** [comment]
Hey @skatcat31 👋 Yea no problem. 

That's a fair question. Hopefully my response helps.

The key word here is demystifying. By demystifying I mean exposing all their methods/function in a way that is easy & simple for the everyday/beginner programmer to understand. As amazing as lodash is, it isn't a simple feat to read through and understand all of lodash methods for your everyday/beginner. Especially when a lot of their method re-uses a lot of their other methods and/or other libraries. The goal here is to have users just find a snippet similar to lodash's snippet and say <strong>*"ahhhh ok I kind of get it"*</strong> or <strong>*"hmmmm I wonder what Array.from() is why not use Array.apply()"*</strong> or even <strong>*"What if I just tweak x or y"*.</strong>

This is very important because it's simple and the barrier to entry is low. Someone just getting started can understand our snippets fairly easily. So why not expose them to one of the most used methods in our industry?

Think of demystifying lodash as a way to creating a beginners guide to lodash methods or understanding lodash guide in 30 seconds or less. I definitely think it not only adds value for Non Traditional who don't have a CS degree or are switching from a dying industry and trying to learn/understand new methods they have or haven't used. But also adds value for the kids in elementary/middle/high school who have just finished or started a codeacademy JavaScript course. None the less I don't think continuously working on the demystifying lodash backlog negatively impacts the projects.

@Chalarangelo Thoughts?

**Comment by skatcat31** [comment]
While it present a great learning opportunity, if they're at that level of beginner that they can't understand lodashes documentation and some of it's source code I fail to see how we could make it clearer. For some lodash or other library methods that could use an easier to understand version I totally agree.

**Comment by kingdavidmartins** [comment]
> Maybe instead of demystifying all of lodash, elect which ones to demystify

I def agree with that statement. To date there have been quite a few methods/functions we haven't really touched and have elected to not implement For example because it's natively supported

@skatcat31 You can also proactively look for snippets/methods on the log you think we shouldn't implement. Which the collaborators/community can vote on in agreement or not

**Comment by skatcat31** [comment]
@kingdavidmartins That makes my complaint null if we can vote on what to not do. The question then becomes "How do we vote?"

**Comment by kingdavidmartins** [comment]
@skatcat31 It's now Titled: Elect Lodash Methods To Demystify ~ Log

**Comment by feiwen8772** [comment]
@kingdavidmartins @Chalarangelo I really like this project! This project should be more popular than Lodash. This project is a collection of code snippets, not a library. More simple and easy to learn.

**Comment by kingdavidmartins** [comment]
Hey @feiwen8772 👋 

Aww Thanks 💯 ❤️ This means soooo much. This project is made possible by @Chalarangelo & all the other contributor's that worked so hard making it all possible. You included 😉 

Thanks again 😄

**Comment by Chalarangelo** [comment]
@kingdavidmartins Great explanation on why we are doing this. @skatcat31 my two cents on the matter are as follows:
- Tools like jQuery and Lodash are industry standards and people use them without knowing their inner workings. Many developers on SO for example feel like jQuery is some kind of voodoo magic that cannot be done with regular JS. We need to show them otherwise!
- Lodash has a ton of useful functions that I would want to use in a project, but I avoid adding huge libraries when I can help it and I like learning how their guts work, so I can modify them. Therefore, it provides us with a great opportunity to implement useful and commonly used methods for people who want to understand how it works and want to hack them a little bit, see what they can make of them.
- As far as it has been discussed, we are not going to be building a library out of this, unless demand is extremely high. In that case, we *might* make a secondary project that releases every month with stable and well-tested snippets. Definitely do not bet on that, though.

**TL;DR:** Lodash is a great starting point and new contributors know what to build and what the expected functionality is, plus we know that the methods we are implementing are useful and we do not have to figure out if they are a good fit in term of usability or not.

**About voting:** I believe in community moderation and decisions, so I'm very glad that we can now decide which methods we do not want. For me the rule of thumb has always been that if something is natively implemented, we do not rewrite it (unless we extend functionality). That being said, we have a bunch of `isX()` methods that contradict my statement. Kindly suggest what we should be doing with them in the comments below.

**Comment by mycaule** [comment]
Other libs to consider: https://vocajs.com, https://date-fns.org

With ES6 you can `import` only some of the functions, which allows developer to pull a banana 
from the tree without pulling the whole jungle.

All these libs are working on more ES6-ish implementation.

**Comment by Chalarangelo** [comment]
@mycaule Truth be told we are working on Lodash, as it was the easiest/closest to our scope. Ramda is something that has been mentioned before and we might dissect a few of its methods, while jQuery's demistification is something I personally wanted to work on these past few days.

Overall, we are trying to find as much worthwhile material and convert it to our structure, so that people can learn as much as possible and understand their tools

**Comment by mycaule** [comment]
Lodash isn't mystifying anything, original library underscore.js even has annotated source, and developers could always look directly at source code if any doubt.
http://underscorejs.org/docs/underscore.html

**Comment by skatcat31** [comment]
@mycaule yes they can read through it, but it's often easier to see a simple short example than traversing their 1K+ lines of code in those libraries. I constantly implement features of Ramda in vanilla ES6 because I don't want to have to import the many helper functions they use. While not a hit on performance, I don't want a front end to have to load their entire packages for a single function only to discard most of it. If I can use a packager or a smaller version, I often elect to do  that.

As for the lessons features we do have the Adapter category right at the top which shows some useful functions. We do need to clean up their descriptions though and put compose into that category as well with a description that states why they would use compose over pipe( algebriac transformation order versus lexical reading order  are easier for people to understand based on the context ).

Truth be told once we're done cleaning this up, a longer "lectures"/"examples" namespace might be and interesting extra page of learning materials for people to use.

**Comment by Chalarangelo** [comment]
I think that the `set`, `unset`, `setWith`, `update` and `updateWith` methods in Object are nearly impossible to implement without making assumptions or writing really bad code or code that is hard to explain. I vote to skip those entirely!

**Comment by fejes713** [comment]
@Chalarangelo Agree, plus we don't have to copy the whole Lodash 👍

**Comment by Chalarangelo** [comment]
I think we are reaching a point where we need to decide what we want from the leftover stuff from lodash. There's definitely plenty of stuff we could really do without.

On the subject of the ones I mentioned, I am marking them off in the log so that we can work on others.

**Comment by Chalarangelo** [comment]
After 1.5 months of work, all the worthwhile methods from Lodash have been added as snippets in the repository. Huge thanks to everyone who contributed to this enormous project!
